controllers/storagecluster/external_resources.go

```diff
@@ -390,22 +391,17 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 					return err
 				}
 			} else if d.Name == cephRgwStorageClassName {
-				rgwEndpoint := d.Data[externalCephRgwEndpointKey]
-				if err := checkEndpointReachable(rgwEndpoint, 5*time.Second); err != nil {
-					r.Log.Error(err, "RGW endpoint is not reachable.", "RGWEndpoint", rgwEndpoint)
-					return err
-				}
-				extCephObjectStores, err = r.newExternalCephObjectStoreInstances(instance, rgwEndpoint)
-				if err != nil {
-					return err
-				}
+				rgwEndpoint = d.Data[externalCephRgwEndpointKey]
 				// rgw-endpoint is no longer needed in the 'd.Data' dictionary,
 				// and can be deleted
 				// created an issue in rook to add `CephObjectStore` type directly in the JSON output
 				// https://github.com/rook/rook/issues/6165
 				delete(d.Data, externalCephRgwEndpointKey)
 
-				scc = newCephOBCStorageClassConfiguration(instance)
+				// do not create the rgw storageclass if the endpoint is not reachable
+				if err := checkEndpointReachable(rgwEndpoint, 5*time.Second); err != nil {
+					scc = newCephOBCStorageClassConfiguration(instance)
+				}
```

Comment:
`err != nil` is true when endpoint is not reachable. So wouldn't this create a SC when endpoint is not reachable?

Reply:
Correct it was bug in the code, I did wrote will update the PR and will share the manually and share the testing results